backend/cn/lib/builtins.ml

```diff
@@ -36,10 +35,37 @@ let mk_arg3_err mk args loc =
 
 let mk_arg3 mk = mk_arg3_err (fun tup loc -> return (mk tup loc))
 
-let mk_arg5 mk args loc =
-  match args with
-  | [ a; b; c; d; e ] -> return (mk (a, b, c, d, e) loc)
-  | xs -> fail { loc; msg = Number_arguments { has = List.length xs; expect = 5 } }
+let var_binop op ty (lhs_sym, lhs_t) (rhs_sym, rhs_t) =
+  binop
+    op
+    (sym_ (lhs_sym, lhs_t, builtin_location), sym_ (rhs_sym, rhs_t, builtin_location))
+    builtin_location
+    ty
```

Comment:
We usually use `bt` or `sbt` to name types; you can also use labelled arguments to distinguish left and right, so for example:
```ocaml
let var_binop op ret_bt ~left:(sym1, bt1) ~right:(sym2, bt2) =
```